bootloader/lk.go

```diff
@@ -103,13 +275,42 @@ func (l *lk) GetBootVars(names ...string) (map[string]string, error) {
 	return out, nil
 }
 
+func (l *lk) newenv() (*lkenv.Env, error) {
+	// check which role we are, it affects which struct is used for the env
+	var version lkenv.Version
+	switch l.role {
+	case RoleSole:
+		version = lkenv.V1
+	case RoleRecovery:
+		version = lkenv.V2Recovery
+	case RoleRunMode:
+		version = lkenv.V2Run
+	}
+	f, err := l.envFile()
+	if err != nil {
+		return nil, err
+	}
+
+	backup, err := l.backupEnvFile()
+	if err != nil {
+		return nil, err
+	}
+
+	return lkenv.NewEnv(f, backup, version), nil
+}
+
 func (l *lk) SetBootVars(values map[string]string) error {
-	env := lkenv.NewEnv(l.envFile(), "", lkenv.V1)
-	if err := env.Load(); err != nil && !os.IsNotExist(err) {
+	env, err := l.newenv()
+	if err != nil {
+		return err
+	}
+	// if we couldn't find the env, that's okay, as this may be the first thing
+	// to write boot vars to the env
```

Comment:
this isn't really true though if we are dealing with paritions, no? I think we should check that partition exist in envFile or not ignore the here if we aren't in a file situation?

Reply:
mmm fair point, the file should always exist at runtime, but still unclear if the file could not exist at image build time

Reply:
OTOH the way from kernel event to udev creating a symlink is long enough that errors are not impossible.

Would this work then?
```
if err := env.Load(); err != nil {
    if !xerrors.Is(err, os.ErrNotExist) {
        return err
    }
    // env file or partition was not found
    if !l.prepareImageTime {
        // a portition with lk environment is missing for unknown reasons
        // this is highly unexpected and and we cannot proceed
        return err
    }
}
```

Reply:
@bboozzoo's proposal seems reasonable

Reply:
yes, I went through all the cases where we Load() and thought more about the cases of IsNotExist and whether or not it should be handled, I think really only SetBootVars at prepare-image time is the only place we should be lenient about IsNotExist and proceed anyways. All the other places I have left an appropriate comment about not handling it.